### ISSUE TITLE
Fix false positive on JOIN LATERAL by adding "lateral" to ALLOWED_FROM_CONTEXTS

### ISSUE THREAD

**Issue by tonyhoang-cot** [dbt-checkpoint/dbt-checkpoint#352]
## Describe the bug

`check-script-has-no-table-name` flags `lateral` as a hardcoded table reference when a model uses `INNER JOIN LATERAL ...` (or any `JOIN LATERAL`) syntax. This is valid SQL in Snowflake, Postgres, BigQuery, and Databricks — `LATERAL` is a keyword that introduces a correlated subquery or table function, not a table name.

There is a related closed issue (#7) from 2021 which fixed the comma-stripping case (`from source, lateral flatten(...)`), but that fix didn't add `lateral` itself to any allow-list. The `JOIN LATERAL ...` form still fails.

PR #291 added an `ALLOWED_FROM_CONTEXTS` list precisely for cases like this (it includes `unnest`, `position`, `distinct`), but `lateral` was not added.

## To Reproduce

A minimal dbt model that triggers the false positive:

```sql
with raw as (
    select 1 as id, parse_json('{"items":[1,2,3]}') as payload
)

select
    raw.id,
    u.value::number as item
from raw
inner join
    lateral flatten(input => raw.payload:items) as u
```

`.pre-commit-config.yaml`:

```yaml
repos:
  - repo: https://github.com/dbt-checkpoint/dbt-checkpoint
    rev: v2.0.8
    hooks:
      - id: check-script-has-no-table-name
```

Running pre-commit produces:

```
Check the script has no table name.......................................Failed
- hook id: check-script-has-no-table-name
- exit code: 1

models/example.sql: does not use source() or ref() macros for tables:
 - lateral
```

## Expected behaviour

`lateral` should be recognised as a SQL keyword and not flagged as a hardcoded table reference. The model above should pass the hook.

## Root cause

In `dbt_checkpoint/check_script_has_no_table_name.py`, the walker fires the rule "previous token was `from` or `join` → current token is a table" when it encounters `join lateral`. There is no awareness that `lateral` is a SQL keyword introducing a table-function or correlated subquery.

## Suggested fix

Adding `"lateral"` to `ALLOWED_FROM_CONTEXTS` resolves the case shown above:

```python
ALLOWED_FROM_CONTEXTS: List[str] = ["distinct", "position", "unnest", "lateral"]
```

This mirrors how `unnest` is already handled (BigQuery's analogue of `LATERAL`).

Note that `LATERAL` can appear in two common forms — both should be handled:

1. `JOIN LATERAL <table_function>(...)` — covered by the suggested fix above.
2. `FROM <table>, LATERAL <table_function>(...)` — also covered, since the walker would see `from`/`,` then later `lateral`. The 2021 fix in #11 handled the comma-stripping; this completes the keyword recognition.

## Version

- `dbt-checkpoint`: v2.0.8
- Database adapter: dbt-snowflake (also affects dbt-postgres, dbt-bigquery, dbt-databricks)
